Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt

```diff
@@ -130,11 +137,115 @@ fun IncludeCertificateQrcodeCardBinding.bindValidityViews(
             startValidationCheckButton.isVisible = isPersonDetails
         }
     }
+}
+
+@Suppress("LongParameterList", "ComplexMethod")
+fun PersonOverviewItemBinding.setUIState(
+    primaryCertificate: CwaCovidCertificate,
+    secondaryCertificate: CwaCovidCertificate? = null,
+    colorShade: PersonColorShade,
+    statusBadgeText: Int = 0,
+    badgeCount: Int = 0,
+    onCovPassInfoAction: () -> Unit
+) {
+    val context = root.context
+    val valid = primaryCertificate.isDisplayValid
+    val color = when {
+        valid -> colorShade
+        else -> PersonColorShade.COLOR_INVALID
+    }
+
+    backgroundImage.setImageResource(color.background)
+    starsImage.setImageDrawable(starsDrawable(context, color))
+    name.text = primaryCertificate.fullName
     certificateBadgeCount.isVisible = badgeCount != 0
     certificateBadgeCount.text = badgeCount.toString()
     certificateBadgeText.isVisible = badgeCount != 0
+    qrCodeCard.apply {
+        loadQrImage(primaryCertificate)
+        statusText.isVisible = statusBadgeText != 0
+        statusBadge.isVisible = statusBadgeText != 0
+        if (statusBadgeText != 0) {
+            statusBadge.text = context.getString(statusBadgeText)
+        }
+        covpassInfoTitle.isVisible = valid
+        covpassInfoButton.isVisible = valid
+        covpassInfoButton.setOnClickListener { onCovPassInfoAction() }
+        invalidOverlay.isGone = valid
+        image.isEnabled = valid
+        certificateToggleGroup.isVisible = secondaryCertificate != null
+        primaryCertificateButton.typeface = Typeface.DEFAULT_BOLD
+        secondaryCertificateButton.typeface = Typeface.DEFAULT
+        certificateToggleGroup.setOnCheckedChangeListener { checkedId ->
```

Comment:
Currently the switch effect is working upon user click, which means:
1- When app is set in background and fragment is restored again the user selection is gone 
2- In case many people in the list of person scrolling up and down will also reset user selection.

I suggest that you set user selection per person as part of the ViewModel state as long as the viewModel is alive you will be able to reflect that in the `UI`

Reply:
I was wondering if we should save the selection state for each person but the tech spec doesn't require it yet. 
Saving it just for the lifecycle of the ViewModel is and edge case and probably a handful of people will stumble upon this. If you were to switch screens and come back to the certificates list, the state would be lost anyway. And since we are now limiting the persons list to only 20, I doubt scrolling will reset the selection. I've just tested it with 30 certificates and it still retains the selection.